roles/etcd/tasks/main.yml

```diff
@@ -1,61 +1,74 @@
-
-- name: setup iptables for etcd
-  shell: >
-      ( iptables -L INPUT | grep "{{ etcd_rule_comment }} ({{ item }})" ) || \
-      iptables -I INPUT 1 -p tcp --dport {{ item }} -j ACCEPT -m comment --comment "{{ etcd_rule_comment }} ({{ item }})"
-  become: true
-  with_items:
-    - "{{ etcd_client_port1 }}"
-    - "{{ etcd_client_port2 }}"
-    - "{{ etcd_peer_port1 }}"
-    - "{{ etcd_peer_port2 }}"
-
-#  Save rules into /etc/sysconfig/iptables file for restoring rules on boot 
-- name: Save iptables
-  command: service iptables save
-  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
-
-- name: copy the etcd start/stop script
-  template: src=etcd.j2 dest=/usr/bin/etcd.sh mode=u=rwx,g=rx,o=rx
-
-- name: copy systemd units for etcd
-  copy: src=etcd.service dest=/etc/systemd/system/etcd.service
-
-- name: start etcd
-  systemd: name=etcd daemon_reload=yes state=started enabled=yes
+- block:
+    - name: download etcdctl {{ etcd_version }}
+      get_url:
+        validate_certs: "{{ validate_certs }}"
+        url: https://github.com/coreos/etcd/releases/download/{{ etcd_version }}/etcd-{{ etcd_version }}-linux-amd64.tar.gz
+        dest: /tmp/etcd-{{ etcd_version }}-linux-amd64.tar.gz
+      tags:
+        - prebake-for-dev
 
+    - name: install etcdctl binaries
+      shell: >
+          tar vxzf /tmp/etcd-{{ etcd_version }}-linux-amd64.tar.gz &&
+          mv etcd-{{ etcd_version }}-linux-amd64/etcd* /usr/bin
+      tags:
+        - prebake-for-dev
+
+    - name: pull etcd container {{ etcd_version }}
+      shell: docker pull quay.io/coreos/etcd:{{ etcd_version }}
+      tags:
+        - prebake-for-dev
+
+    # Make Iptables rule persistent across reboot
+    - name: Install iptable-service package
+      yum: pkg={{ item }} state=present
+      with_items:
+      - iptables-services
+      when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+    - name: Install iptables-persistent
+      apt: name={{item}} state=installed
+      with_items:
+          - iptables-persistent
+      when: ansible_distribution == 'Debian' or ansible_distribution == 'Ubuntu'
+
+    - name: setup iptables for etcd
+      shell: >
+          ( iptables -L INPUT | grep "{{ etcd_rule_comment }} ({{ item }})" ) || \
+          iptables -I INPUT 1 -p tcp --dport {{ item }} -j ACCEPT -m comment --comment "{{ etcd_rule_comment }} ({{ item }})"
+      become: true
+      with_items:
+        - "{{ etcd_client_port1 }}"
+        - "{{ etcd_client_port2 }}"
+        - "{{ etcd_peer_port1 }}"
+
+    #  Save rules into /etc/sysconfig/iptables file for restoring rules on boot
+    - name: Save iptables
+      command: service iptables save
+      when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+    - name: Set facts from role defaults for sharing with other hosts
+      set_fact:
+        etcd_node_name: "{{ etcd_node_name }}"
```

Comment:
would be helpful having `contiv` prefix, or just `contiv_etcd{0,3}`

Reply:
could you elaborate, I'm not sure what you are asking or the benefit of namespacing here, it's not something end user interacts with

Reply:
not when they don't have to debug it ... indicate this is etcd for contiv instead of other general purpose etcd